MatrixGame/src/MatrixGameDll.cpp

```diff
@@ -164,167 +69,28 @@ int __stdcall Support() {
     return SUPE_OK;
 }
 
-// 0-exit to windows (terminate)
-// 1-exit to main menu
-// 2-loss
-// 3-win
-
-void protectedBlock1(CFormMatrixGame *formgame) {
-    try {
-        g_ExitState = 0;
-        FormChange(formgame);
+int __stdcall Run(HINSTANCE hinst, HWND hwnd, wchar *map, SRobotsSettings *set, wchar *lang, wchar *txt_start,
+                  wchar *txt_win, wchar *txt_loss, wchar *planet, SRobotGameState *rgs) {
 
-        timeBeginPeriod(1);
+    CGame game{};
 
-        SETFLAG(g_Flags, GFLAG_APPACTIVE);
-        RESETFLAG(g_Flags, GFLAG_EXITLOOP);
+    uint32_t seed = (unsigned)time(NULL);
 
-        L3GRun();
-    }
-    catch (... /*ExceptionHandler(GetExceptionInformation())*/) {
-        SETFLAG(g_Flags, GFLAG_EXITLOOP);
-        g_ExitState += 100;
-    }
-}
-
-void protectedBlock2(CFormMatrixGame *formgame, SRobotGameState *rgs) {
-    try {
-        rgs->m_Time = g_MatrixMap->GetPlayerSide()->GetStatValue(STAT_TIME);
-        rgs->m_BuildRobot = g_MatrixMap->GetPlayerSide()->GetStatValue(STAT_ROBOT_BUILD);
-        rgs->m_KillRobot = g_MatrixMap->GetPlayerSide()->GetStatValue(STAT_ROBOT_KILL);
-        rgs->m_BuildTurret = g_MatrixMap->GetPlayerSide()->GetStatValue(STAT_TURRET_BUILD);
-        rgs->m_KillTurret = g_MatrixMap->GetPlayerSide()->GetStatValue(STAT_TURRET_KILL);
-        rgs->m_KillBuilding = g_MatrixMap->GetPlayerSide()->GetStatValue(STAT_BUILDING_KILL);
-    }
-    catch (...) {
-    }
+    game.Init(hinst, hwnd, map, seed, set, lang, txt_start, txt_win, txt_loss, planet);
+    
+    CFormMatrixGame *formgame = HNew(NULL) CFormMatrixGame();
```

Comment:
seems like you don't need `new` call here, just create a regular object.

Reply:
Right